### PR TITLE
Upgrade rescript tree sitter & highlights

### PR DESCRIPTION
**Pull request by jaredramirez**
Hey there, this PR updates the rescript tree sitter and it's corresponding queries!

### CODE REVIEW

File: runtime/queries/rescript/highlights.scm [hunk 1]
```diff
@@ -10,20 +10,28 @@
...
-  "list"
 ] @type
 
+
+(list "list{" @type)
+(list_pattern "list{" @type)
+
+; To ensure that the closing curly bracket is the same color (scope) as the opening curly bracket
+(list "}" @type (#set! "priority" 105))
+(list_pattern "}" @type (#set! "priority" 105))
```
Comment (the-mikedavis) [inline runtime/queries/rescript/highlights.scm:21]: We don't implement a `priority` property so the `(#set! ..)`s here can be removed. These blocks can also be combined as long as `list` and `list_pattern` always have `}` nodes:

```scm
(list ["list{" "}"] @type)
(list_pattern ["list{" "}"] @type)
```

File: runtime/queries/rescript/highlights.scm [hunk 2]
```diff
@@ -74,47 +84,65 @@
  "@"
  "@@"
  (decorator_identifier)
-] @label
+] @keyword.directive
 
 (extension_identifier) @keyword
-("%") @keyword
+("%") @keyword.
```
Comment (the-mikedavis) [inline runtime/queries/rescript/highlights.scm:90]: Looks like this may be a typo?

File: runtime/queries/rescript/textobjects.scm [hunk 3]
```diff
@@ -1,16 +1,114 @@
+; Queries for nvim-treesitter/nvim-treesitter-textobjects
```
Comment (the-mikedavis) [inline runtime/queries/rescript/textobjects.scm:1]: This line could be removed or could point to the upstream source for these queries if there is one

File: runtime/queries/rescript/textobjects.scm [hunk 4]
```diff
@@ -1,16 +1,114 @@
...
+; Calls
+;------
+
+(call_expression arguments: ((_) @parameter.inside)) @parameter.around
+
 ; Comments
 ;---------
 
-(comment) @comment.inside
+(comment) @comment.around
```
Comment (the-mikedavis) [inline runtime/queries/rescript/textobjects.scm:27]:
```suggestion
(comment) @comment.inside
(comment)+ @comment.around
```

**Comment by jaredramirez** [comment]
Thanks for the comments, I've committed the updates!

**Review by the-mikedavis — APPROVED** [review]

**Comment by the-mikedavis** [comment]
Thanks!

**Pull request merged**